src/blockdata/witness.rs

```diff
@@ -282,8 +292,36 @@ impl<'de> serde::Deserialize<'de> for Witness {
     where
         D: serde::Deserializer<'de>,
     {
-        let vec: Vec<Vec<u8>> = serde::Deserialize::deserialize(deserializer)?;
-        Ok(Witness::from_vec(vec))
+        struct Visitor;
+        impl<'de> serde::de::Visitor<'de> for Visitor
+        {
+            type Value = Witness;
+
+            fn expecting(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
+                write!(f, "a sequence of hex arrays")
+            }
+
+            fn visit_seq<A: serde::de::SeqAccess<'de>>(self, mut a: A)
+                                                       -> Result<Self::Value, A::Error>
```

Comment:
Can we have this all as a single line? The indentation is unusual for the rest of the codebase, also for what its worth, on the same line its 105 characters and on line 311 we have 118 characters, so its not excessively long.

Reply:
This is actually a copy-paste from other place exactly in this codebase :) But sure I'll fix the identation. Just never pay attention to it since using `rustfmt` in all other projects and the code formats itself.

Reply:
oh drats, I would have hoped my 'manual rustfmt fixes' PR would have caught that.